File: src/main/java/br/com/blu/bs2/sintetizador/contas/csv/model/dominio/ExtratoConta.java
package br.com.blu.bs2.sintetizador.contas.csv.model.dominio;

import br.com.blu.bs2.sintetizador.contas.csv.service.strategy.PercentualBonus;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.util.Objects;

@Data
@Builder
public class ExtratoConta {
    private Conta conta;
    private BigDecimal valorTotalDeposito;
    private BigDecimal valorPercentualBonus;

    public void aplicarValorPercentualBonus(PercentualBonus percentualBonus) {
        if (Objects.isNull(this.conta)) {
            this.valorPercentualBonus = BigDecimal.ZERO;
            return;
        }
        this.valorPercentualBonus = (Objects.isNull(percentualBonus))
                ? BigDecimal.ZERO
                : percentualBonus.getFactoryObterPercentualBonusService().obterValorPercentualBonusAoDeposito(this.conta);
    }
}

File: src/main/kotlin/br/com/blu/bs2/sintetizador/contas/csv/model/dominio/ExtratoConta.kt
package br.com.blu.bs2.sintetizador.contas.csv.model.dominio

import br.com.blu.bs2.sintetizador.contas.csv.service.strategy.PercentualBonus
import java.math.BigDecimal
import java.util.*

class ExtratoConta {
    var conta: Conta? = null
    var valorTotalDeposito: BigDecimal? = null
    var valorPercentualBonus: BigDecimal? = null

    constructor() {}
    constructor(conta: Conta?, valorTotalDeposito: BigDecimal?, valorPercentualBonus: BigDecimal?) {
        this.conta = conta
        this.valorTotalDeposito = valorTotalDeposito
        this.valorPercentualBonus = valorPercentualBonus
    }

    fun aplicarValorPercentualBonus(percentualBonus: PercentualBonus) {
        if (Objects.isNull(conta)) {
            valorPercentualBonus = BigDecimal.ZERO
            return
        }
        valorPercentualBonus = if (Objects.isNull(percentualBonus)) BigDecimal.ZERO else percentualBonus.factoryObterPercentualBonusService.obterValorPercentualBonusAoDeposito(conta)
    }

    override fun toString(): String {
        return "ExtratoConta{" +
                "conta=" + conta +
                ", valorTotalDeposito=" + valorTotalDeposito +
                ", valorPercentualBonus=" + valorPercentualBonus +
                '}'
    }
}
